packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -3,18 +3,77 @@ pragma solidity 0.8.15;
 
 import { IPreimageOracle } from "./interfaces/IPreimageOracle.sol";
 import { PreimageKeyLib } from "./PreimageKeyLib.sol";
+import { LibKeccak } from "@lib-keccak/LibKeccak.sol";
 import "./libraries/CannonErrors.sol";
+import "./libraries/CannonTypes.sol";
 
 /// @title PreimageOracle
 /// @notice A contract for storing permissioned pre-images.
+/// @custom:attribution Solady <https://github.com/Vectorized/solady/blob/main/src/utils/MerkleProofLib.sol#L13-L43>
+/// @custom:attribution Beacon Deposit Contract / <0x00000000219ab540356cbb839cbe05303d7705fa>
 contract PreimageOracle is IPreimageOracle {
+    ////////////////////////////////////////////////////////////////
+    //                         Constants                          //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice The depth of the keccak256 merkle tree. Supports up to 65,536 keccak blocks, or ~8.91MB preimages.
+    uint256 public constant KECCAK_TREE_DEPTH = 16;
+    /// @notice The maximum number of keccak blocks that can fit into the merkle tree.
+    uint256 public constant MAX_LEAF_COUNT = 2 ** KECCAK_TREE_DEPTH - 1;
+    /// @notice The duration of the large preimage proposal challenge period.
+    uint256 public constant CHALLENGE_PERIOD = 1 days;
```

Comment:
Are we comfortable with this being a constant and not parameterized by another part of the fault proof system?

Reply:
I think that this is fine. We plan on the `PreimageOracle` being an immutable contract. We can always replace it by upgrading the `FaultDisputeGame` as well :)

Reply:
And can make it configurable during the bond spike if need be.

Reply:
Agreed, shouldn't be a blocker in the future so let's ship it